examples/cat_facts/shared/src/app.rs

```diff
@@ -136,62 +140,82 @@ impl App for CatFacts {
             Event::Fetch => {
                 model.cat_image = Some(CatImage::default());
 
-                caps.http
+                let eff1 = caps
+                    .http
                     .get(FACT_API_URL)
                     .expect_json()
-                    .send(Event::SetFact);
+                    .send_and_respond(Event::SetFact);
 
-                caps.http
+                let eff2 = caps
+                    .http
                     .get(IMAGE_API_URL)
                     .expect_json()
-                    .send(Event::SetImage);
+                    .send_and_respond(Event::SetImage);
 
-                caps.render.render();
+                let eff3 = caps.render.render();
+                eff1.join(eff2).join(eff3)
             }
             Event::SetFact(Ok(mut response)) => {
                 model.cat_fact = Some(response.take_body().unwrap());
 
                 let bytes = serde_json::to_vec(&model).unwrap();
-                caps.key_value.set(KEY.to_string(), bytes, |_| Event::None);
-
-                caps.time.now(Event::CurrentTime);
+                Command::effect(async move {
+                    caps.key_value.set_async(KEY.to_string(), bytes).await;
+                    Command::none()
+                })
+                .join(caps.time.now(Event::CurrentTime))
             }
             Event::SetImage(Ok(mut response)) => {
                 model.cat_image = Some(response.take_body().unwrap());
 
                 let bytes = serde_json::to_vec(&model).unwrap();
-                caps.key_value.set(KEY.to_string(), bytes, |_| Event::None);
 
-                caps.render.render();
+                Command::effect(async move {
+                    let _result = caps.key_value.set_async(KEY.to_string(), bytes).await;
+                    Command::none()
+                })
+                .join(async move {
```

Comment:
presumably this could just be
```rust
.join(caps.render.render())
```

or is this just to show that either is valid?

Reply:
I have reworked the Commands and effects to make this cleaner